pkgs/tools/inputmethods/fcitx5/with-addons.nix

```diff
@@ -20,12 +22,18 @@ symlinkJoin {
     fcitx5-configtool
   ] ++ addons;
 
+  buildInputs = [
+    gdk-pixbuf
```

Comment:
gdk-pixbuf will bring gnome into the dependencies, which isn't used by everyone. The desktop special one should not appear in the generic wrapper. Adding a new nixos option under fcitx5 module, configuring this in fcitx5 with checking desktop environment or documenting how to set it with system.environment are better methods.

Reply:
Yeah, thinking about it you're right. Tbh, I'm still not completely clear on this issue, because `$GDK_PIXBUF_MODULE_FILE` is not even set on my system. 

But when `fcitx5` is run outside of a service, it just works anyways. So I guess it's being set somewhere, and that's somehow not being triggered inside the service.

For now I'm adding this to my home-manager:

```nix
home.sessionVariables =
  let
    loadersCache = pkgs.gnome._gdkPixbufCacheBuilder_DO_NOT_USE {
      extraLoaders = [ pkgs.librsvg ];
    };
  in
  {
    GDK_PIXBUF_MODULE_FILE = "${loadersCache}";
  };
```

It's not a very nix-ish thing to do, but I guess it's fine for now.

Reply:
> that's somehow not being triggered inside the service.

Does the service mean as a systemd service?

The systemd service only includes same basic environment without the desktop ones.  By the way, if you use gnome, gnome can use xdg-autostart to start fcitx5, which should include these environment.